webapp/src/components/gitlab_issue_selector.tsx

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState, useEffect, useCallback} from 'react';
+import {Theme} from 'mattermost-redux/types/preferences';
+
+import debounce from 'debounce-promise';
+import AsyncSelect from 'react-select/async';
+import {SingleValue} from 'react-select';
+
+import {getStyleForReactSelect} from 'src/utils/styles';
+import Client from 'src/client';
+import Setting from './setting';
+
+const searchDebounceDelay = 400;
+
+interface PropTypes {
+    name: string;
+    required: boolean;
+    theme: Theme;
+    onChange: (value: Issue | null) => void;
+    error: string;
+    value: Issue | null;
+    addValidate: (key: string, validateField: () => boolean) => void;
+    removeValidate: (key: string) => void;
+};
+
+const GitlabIssueSelector = ({name, required, theme, onChange, error, value, addValidate, removeValidate
+}: PropTypes) => {
+    const [invalid, setInvalid] = useState(false);
+    const [responseError, setResponseError] = useState('')
+
+    const isValid = useCallback(() => {                
+        if (!required) {
+            return true;
+        }
+        
+        const valid = Boolean(value);
+        setInvalid(!valid);
+        return valid;
+    }, [value, required])
+
+    useEffect(() => {
+        return () => {            
+            if (removeValidate && name) {
+                removeValidate(name);
+            }
+        }
+    }, [])
+
+    useEffect(() => {
+        if (addValidate && name) {            
+            addValidate(name, isValid);
+        }
+        if (invalid) {            
+            isValid();
+        }
+    }, [isValid])
+
+    const handleIssueSearchTermChange = useCallback((inputValue: string) => {
+        return debouncedSearchIssues(inputValue);
+    }, []);
+
+    const searchIssues = useCallback(async (text: string) => {
+        const textEncoded = encodeURIComponent(text.trim().replace(/\\/g, '\\\\').replace(/"/g, '\\"'));
+        try {
+            const issues = await Client.searchIssues(textEncoded);
+
+            if (!Array.isArray(issues)) {
+                return [];
+            }
+
+            return issues.map((issue) => {
+                const projectParts = issue.web_url.split('/');
+                let prefix = '';
+                // Extract "username/projectName" from the issueURL parts
+                if (projectParts.length >= 5) {
+                    prefix = `${projectParts[projectParts.length - 5]}/${projectParts[projectParts.length - 4]}`;
+                }
```

Comment:
Is it possible that the URL doesn't match what want to to here? Maybe there are somtimes other suffixes on the URL that make this `length - 4` etc. thing not work properly?

Reply:
Now we are traversing the array from starting index

Reply:
Can you write a unit test for this functionality?

Reply:
@mickmister I did the setup to write unit tests in [PR: #301](https://github.com/mattermost/mattermost-plugin-gitlab/pull/301). So should I wait for that PR to be merged or do setup here as well and write unit tests?

Reply:
@Nityanand13 I suppose you'll need to port the setup here as well, as I'm not sure when the other PR will be merged